src/nifreeze/utils/iterators.py

```diff
@@ -33,14 +33,17 @@ def linear_iterator(size: int | None = None, **kwargs) -> Iterator[int]:
 
     Parameters
     ----------
-    size : :obj:`int`
-        Number of volumes in the dataset
-        (for instance, the number of orientations in a DWI)
+    size : :obj:`int` or ``None`` (optional)
+        Number of volumes in the dataset.
```

Comment:
Explain the behavior when value is ``None``?

Reply:
```suggestion
        Number of volumes in the dataset.
        If ``None``, ``size`` will be inferred from the ``bvalues``
        keyword argument.
```